library.js

```diff
@@ -310,7 +316,7 @@ plugin.updateUserProfile = function(uid, userData, isNewUser, callback) {
 
 plugin.updateUserGroups = function (uid, userData, isNewUser, callback) {
 	if (!userData.groups || !userData.groups.length) {
-		return setImmediate(callback, null, uid);
+		return setImmediate(callback, null, uid, isNewUser);
```

Comment:
Any particular reason this is needed in the new hook payload? I assume this is why it is passed all the way down the chain?

Reply:
For my use case I only need to do the checks for new users, so I pass this through. I dont need it for my use case but makes sense to me to pass userData too as the same args are passed through updateProfile, updateGroups and verify then.